### PR TITLE
ci/ui: fix upgrade tests

### LINKED ISSUE #851: e2e UI: Upgrade tests broken
Screenshot 2023-05-31 at 14 27 10

Upgrade tests through UI do not work anymore because first we install stable operator with latest dev UI and it's not compatible anymore because the image is not hardcoded anymore. That means we can not use the Build ISO function anymore.

### PR DESCRIPTION
**Pull request by juadk**
Fix #851

## Verification runs
[UI-K3s-OS-Upgrade-Rancher_Stable](https://github.com/rancher/elemental/actions/runs/5138039358/jobs/9246928859) ✔️ 
[UI-RKE2-OS-Upgrade-Rancher_Stable](https://github.com/rancher/elemental/actions/runs/5143153835/jobs/9257833216) ✔️ 

[UI-K3s-OS-Upgrade-Rancher_Latest](https://github.com/rancher/elemental/actions/runs/5141983111/jobs/9255161667) ✔️ 
[UI-RKE2-OS-Upgrade-Rancher_Latest](https://github.com/rancher/elemental/actions/runs/5144147327/jobs/9260086806) ✔️ 

[UI-RKE2-Rancher_Stable](https://github.com/rancher/elemental/actions/runs/5143920162/jobs/9259557697) ✔️ 
[UI-K3s-Rancher_Stable](https://github.com/rancher/elemental/actions/runs/5141262343/jobs/9253568692) ✔️ 

I also added more memory to the VM if we test UI, tests look more reliable and it does not impact anything because we use only one VM in UI tests.

### CODE REVIEW

File: tests/scripts/install-vm [hunk 1]
```diff
@@ -9,7 +9,8 @@ ARCH=$(uname -m)
 FW_CODE=/usr/share/qemu/ovmf-${ARCH}-smm-suse-code.bin
 FW_VARS=$(realpath ../assets/ovmf-template-vars.fd)
 EMULATED_TPM="none"
-VM_MEM=3072
+# Give more memory to VM if UI is tested (only one VM for now)
+[[ $3 = "UI" ]] && VM_MEM=8192 || VM_MEM=3072
```
Comment (ldevulder) [inline tests/scripts/install-vm]: should be `==` and not `=`.

File: tests/scripts/start-cypress-tests [hunk 2]
```diff
@@ -40,14 +40,15 @@ docker run -v $PWD:/workdir -w /workdir                     \
...
+  # We need to build the ISO with stable base image
+  # It can not be done from the UI for now
+  
   # RKE2 as upstream cluster
   # We cannot use PXE boot so we have to boot from ISO but we cannot use stable ISO
   # We need to use a custom ISO and the menu doesn't allow to do it now.
-  elif grep rke <<< ${K8S_UPSTREAM_VERSION}; then
+  if [[ ${CYPRESS_TAGS} == "upgrade" ]] || grep rke <<< ${K8S_UPSTREAM_VERSION}; then
+    # Wait a bit to let the ISO be built (could be improved)
+    sleep 180
```
Comment (ldevulder) [inline tests/scripts/start-cypress-tests:51]: How us the ISO built? Maybe checking the filesize change can be an option? But anyway for another PR.

**Review by ldevulder — CHANGES_REQUESTED** [review]
LGTM, just a small fix to do.

File: tests/scripts/start-cypress-tests (hunk 2 above)
Comment (juadk) [reply tests/scripts/start-cypress-tests:51]: I agree but this part of code will be removed very soon (when we will get the next UI release)

Comment (ldevulder) [reply tests/scripts/start-cypress-tests:51]: In that case clearly nothing more to do ;-)

**Review by ldevulder — APPROVED** [review]

**Pull request merged**